carsus/io/output/base.py

```diff
@@ -434,6 +438,117 @@ def _create_levels_lines(self, lines_loggf_threshold=-3,
+
+        # Calculate g_ratio
+        collisions["g_ratio"] = collisions["g_l"] / collisions["g_u"]
+
+        # TODO: pass temperatures as a parameter
+        temperatures = np.arange(2000, 50000, 2000)
+
+        # Derive columns for collisional strenghts
+        c_ul_temperature_cols = ['t{:06d}'.format(t) for t in temperatures]
+
+        collisions = collisions.rename(columns={'ion_charge': 'ion_number',
+                                                'temperatures': 'btemp',
+                                                'collision_strengths': 'bscups'})
+        collisions = collisions[['e_col_id', 'lower_level_id',
+                                 'upper_level_id', 'ds_id',
+                                 'btemp', 'bscups', 'ttype', 'cups',
+                                 'gf', 'atomic_number', 'ion_number',
+                                 'level_number_lower', 'g_l',
+                                 'energy_lower', 'level_number_upper', 
+                                 'g_u', 'energy_upper', 'delta_e', 
+                                    'g_ratio']]
+
+        def calculate_collisional_strength(row, temperatures):
+            """
+                Function to calculation upsilon from Burgess & Tully 1992 (TType 1 - 4; Eq. 23 - 38)
+            """
+            c = row["cups"]
+            x_knots = np.linspace(0, 1, len(row["btemp"]))
+            y_knots = row["bscups"]
+            delta_e = row["delta_e"]
+            g_u = row["g_u"]
+
+            ttype = row["ttype"]
+            if ttype > 5: ttype -= 5
+
+            kt = kb_ev * temperatures
+
+            spline_tck = interpolate.splrep(x_knots, y_knots)
+
+            if ttype == 1:
+                x = 1 - np.log(c) / np.log(kt / delta_e + c)
+                y_func = interpolate.splev(x, spline_tck)
+                upsilon = y_func * np.log(kt / delta_e + np.exp(1))
+
+            elif ttype == 2:
+                x = (kt / delta_e) / (kt / delta_e + c)
+                y_func = interpolate.splev(x, spline_tck)
+                upsilon = y_func
+
+            elif ttype == 3:
+                x = (kt / delta_e) / (kt / delta_e + c)
+                y_func = interpolate.splev(x, spline_tck)
+                upsilon = y_func / (kt / delta_e + 1)
+
+            elif ttype == 4:
+                x = 1 - np.log(c) / np.log(kt / delta_e + c)
+                y_func = interpolate.splev(x, spline_tck)
+                upsilon = y_func * np.log(kt / delta_e + c)
+
+            elif ttype == 5:
+                raise ValueError('Not sure what to do with ttype=5')
+
+            #### 1992A&A...254..436B Equation 20 & 22 #####
+
+            c_ul = 8.63e-6 * upsilon / (g_u * temperatures**.5)
+
+            return pd.Series(data=c_ul, index=c_ul_temperature_cols)
+
+        collisional_strengths = collisions.apply(calculate_collisional_strength, axis=1, args=(temperatures,))
```

Comment:
This is not a collision strength.
``` 
c_ul = 8.63e-6 * upsilon / (g_u * temperatures**.5)
```
would be the collisional de-excitation rate coefficient (if we multiply it with the electron density); upsilon is the collision strength. We should think about renaming this for clarity.

Reply:
I don't remember which name we agreed to use @chvogl.

Reply:
Can you change this to `collisional_ul_factor`(which I think was Wolfgang's preferred option)?

Reply:
Done.